org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/outline/ModelBuilder.scala

```diff
@@ -41,11 +41,16 @@ object ModelBuilder extends HasLogger {
     }
 
     def updateTree(parent: ContainerNode, t: Tree): Unit = {
+      def renderArgs[A](sb: StringBuilder, args: List[A])(f: A => Unit) = {
+        if (args.nonEmpty) {
+          f(args.head)
+          args.drop(1).foreach { a => sb.append(", "); f(a) }
```

Comment:
I'm not so perceivable :(. Actually this code looks little strange. Now I see `f` should use same instance of `sb` but is not explicitly expressed. I'm aware you are constrained by existing implementation....